pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -577,16 +577,15 @@ func (c *Controller) HasSynced() bool {
 }
 
 func (c *Controller) informersSynced() bool {
-	if !c.namespaces.HasSynced() ||
-		!c.services.HasSynced() ||
-		!c.endpoints.slices.HasSynced() ||
-		!c.pods.pods.HasSynced() ||
-		!c.nodes.HasSynced() ||
-		!c.imports.HasSynced() ||
-		!c.exports.HasSynced() {
-		return false
-	}
-	return true
+	return c.namespaces.HasSynced() &&
+		c.services.HasSynced() &&
+		c.endpoints.slices.HasSynced() &&
+		c.pods.pods.HasSynced() &&
+		c.nodes.HasSynced() &&
+		c.imports.HasSynced() &&
+		c.exports.HasSynced() &&
+		c.client.CrdWatcher().HasSynced() &&
```

Comment:
I don't think we need to check crdwatcher

Reply:
Didn't realize the delayedClient baked a HasSynced check on CrdWatcher. `kclient` is pretty nice :)